### PR TITLE
f-strings everywhere

### PR DESCRIPTION
**Pull request by taoufik07**
except https://github.com/encode/starlette/blob/master/starlette/middleware/sessions.py#L58 and https://github.com/encode/starlette/blob/master/starlette/middleware/sessions.py#L68

### CODE REVIEW

File: starlette/middleware/wsgi.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def build_environ(scope: Scope, body: bytes) -> dict:
         elif name == "content-type":
             corrected_name = "CONTENT_TYPE"
         else:
-            corrected_name = "HTTP_%s" % name.upper().replace("-", "_")
+            corrected_name = f"HTTP_{name.upper().replace("-", "_")}"
```
Comment (lovelydinosaur) [inline starlette/middleware/wsgi.py]: This isn't valid syntax.  (See test failure... https://travis-ci.org/encode/starlette/jobs/482660913)

File: starlette/datastructures.py [hunk 2]
```diff
@@ -115,8 +115,8 @@ def replace(self, **kwargs: typing.Any) -> "URL":
             if username is not None:
```
Comment (lovelydinosaur) [inline starlette/datastructures.py:115]: Missing `netloc += ":%d" % port`

File: starlette/routing.py [hunk 3]
```diff
@@ -103,16 +103,16 @@ def compile_path(
...
+        ), f"Unknown path convertor '{convertor_type}'"
         convertor = CONVERTOR_TYPES[convertor_type]
 
         path_regex += path[idx : match.start()]
-        path_regex += "(?P<%s>%s)" % (param_name, convertor.regex)
+        path_regex += f"(?P<{param_name}>{convertor.regex})"
 
         path_format += path[idx : match.start()]
-        path_format += "{%s}" % param_name
+        path_format += f"{{{param_name}}}"
```
Comment (lovelydinosaur) [inline starlette/routing.py]: This ends up being non-obvious. I'd probably rather we use `"{" + param_name + "}"` or `netloc += ":%d" % port`

File: starlette/middleware/wsgi.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ def build_environ(scope: Scope, body: bytes) -> dict:
         elif name == "content-type":
             corrected_name = "CONTENT_TYPE"
         else:
-            corrected_name = "HTTP_%s" % name.upper().replace("-", "_")
+            corrected_name = f"HTTP_{name.upper().replace('-', '_')}"
```
Comment (lovelydinosaur) [inline starlette/middleware/wsgi.py]: Let's perhaps pull `.upper().replace('-', '_')` out of the f-string entirely. So we have:

`f"HTTP_{name}".upper().replace("-", "_")`

I think that ends up being cleaner. (I don't really want us to use expressions inside f-strings if we can avoid it.)

**Comment by lovelydinosaur** [comment]
Does also mean we'd have to pass on https://github.com/encode/starlette/issues/187

**Comment by lovelydinosaur** [comment]
Which, yeah, I've currently said we should probably do anyway, but I *do* still like the idea of being able to run starlette on a raspberry pi easily, since it's a great target for it.

**Comment by em92** [comment]
> Does also mean we'd have to pass on #187

> I do still like the idea of being able to run starlette on a raspberry pi easily

I suggest to create separate python3.5 branch or repository for that. There you can revoke those async generators without decreasing readability or adding extra dependencies.

**Comment by em92** [comment]
> it's too difficult to keep in sync.

Merging from master and rebasing to exclude/modify incompatible commits for python 3.5 does not seem to be difficult. I didn't try in practice myself, but I have seen project, that was doing this. As example: https://github.com/brugal/wolfcamql/ merges commits from https://github.com/ioquake/ioq3 (See commit messages containing "ioquake3 patches")

**Comment by lovelydinosaur** [comment]
Let's have this.
Great stuff, thanks! :)

**Pull request merged**